core/src/utils/forms/form-controller.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Creates a controller that tracks whether a form control is using the legacy or modern syntax. This should be removed when the legacy form control syntax is removed.
+ *
+ * @internal
+ * @prop el: The Ionic form component to reference
+ */
+export const createLegacyFormController = (el: HTMLElement): LegacyFormController => {
+  const controlEl: HTMLElement = el;
+  let legacyControl = true;
+
+  /**
+   * Detect if developers are using the legacy form control syntax
+   * so a deprecation warning is logged. This warning can be disabled
+   * by either using the new `label` property or setting `aria-label`
+   * on the control.
+   */
+  const hasLabelProp = (controlEl as any).label !== undefined;
```

Comment:
Instead of casting to `any` here, should we instead add the additional types we are expecting from our custom element in the `controlEl` declaration? 

e.g.:
```ts
const controlEl: HTMLElement & { label?: string } = el;
```

Alternatively, is there an existing type that already captures this vs. using `HTMLElement`?